lib/plugin/vim-plugin.nix

```diff
@@ -0,0 +1,87 @@
+{
+  lib,
+  nixvimOptions,
+  nixvimUtils,
+  mkPlugin,
+}:
+with lib;
+{
+  mkVimPlugin =
+    config:
+    {
+      name,
+      # deprecations
+      deprecateExtraConfig ? false,
+      # colorscheme
+      isColorscheme ? false,
+      colorscheme ? name,
+      # options
+      settingsOptions ? { },
+      settingsExample ? null,
+      globalPrefix ? "",
+      extraOptions ? { },
+      # config
+      extraConfig ? cfg: { },
+      ...
+    }@args:
+    mkPlugin config (
+      let
+        cfg = config.${namespace}.${name};
+
+        globals = cfg.settings or { };
+
+        createSettingsOption = (isString globalPrefix) && (globalPrefix != "");
+
+        namespace = if isColorscheme then "colorschemes" else "plugins";
+
+        settingsOption = optionalAttrs createSettingsOption {
+          settings = nixvimOptions.mkSettingsOption {
+            options = settingsOptions;
+            example = settingsExample;
+            description = ''
+              The configuration options for **${name}** without the `${globalPrefix}` prefix.
+
+              Example: To set `${globalPrefix}foo_bar` to `1`, write
+              ```nix
+                settings = {
+                  foo_bar = true;
+                };
+              ```
+            '';
+          };
+        };
+      in
+      (removeAttrs args [
+        "deprecateExtraConfig"
+        "extraConfig"
+        "extraOptions"
+        "globalPrefix"
+        "imports"
+        "settingsExample"
+        "settingsOptions"
+      ])
+      // {
+        imports =
+          let
+            basePluginPath = [
+              namespace
+              name
+            ];
+            settingsPath = basePluginPath ++ [ "settings" ];
+          in
+          (args.imports or [ ])
+          ++ (optional (deprecateExtraConfig && createSettingsOption) (
+            mkRenamedOptionModule (basePluginPath ++ [ "extraConfig" ]) settingsPath
+          ));
```

Comment:
Isn't this the same as nvim's deprecate extraConfig? Could you just pass `deprecateExtraConfig = deprecateExtraConfig && createSettingsOption` through to mkPlugin?

EDIT: The option name is different too (extraOptions vs extraConfig). I guess not worth it then.

Reply:
> EDIT: The option name is different too (extraOptions vs extraConfig). I guess not worth it then.

indeed...